fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/RelationalExpressionWithPredicates.java

```diff
@@ -49,15 +50,29 @@
      */
     @Nonnull
     static ImmutableSet<FieldValue> fieldValuesFromPredicates(@Nonnull final Collection<? extends QueryPredicate> predicates) {
+        return fieldValuesFromPredicates(predicates, queryPredicate -> true);
+    }
+
+    /**
+     * Return all {@link FieldValue}s contained in the predicates handed in.
+     * @param predicates a collection of predicates
+     * @param predicatePredicate an actual predicate performing additional filtering for the kinds of
+     *        {@link PredicateWithValue}s the caller is interested in
+     * @return a set of {@link FieldValue}s
+     */
+    @Nonnull
+    static ImmutableSet<FieldValue> fieldValuesFromPredicates(@Nonnull final Collection<? extends QueryPredicate> predicates,
+                                                              @Nonnull final Predicate<PredicateWithValue> predicatePredicate) {
```

Comment:
Perhaps `predicateFilter`?

Reply:
`filteringPredicate` as it is a `Predicate` after all. No strong view show if you think this is still not addressing the intent of your comment.